File: 23000s/23335.cpp
#include <bits/stdc++.h>
using namespace std;

int f(int n) {
    int a = 1;
    for(int i=2 ; i*i<=n ; i++) {
        if(i * i == n) {
            a += i;
            break;
        }
        if(n % i == 0) {
            a += i;
            a += n/i;
        }
    }
    return a;
}

int main()
{
    cin.tie(0)->sync_with_stdio(0);
    int n;
    cin >> n;
    while(n--) {
        int a;
        cin >> a;
        if(a == 1) {
            cout << "deficient\n";
            continue;
        }
        int b = f(a);
        if(b > a) cout << "abundant\n";
        else if(b < a) cout << "deficient\n";
        else cout << "perfect\n";
    }
    
}

File: 23000s/23335.py
def s(n):
    a = 1
    i = 2
    while i * i <= n:
        if i * i == n:
            a += i
            break
        if n % i == 0:
            a += i
            a += n // i
        i += 1
    return a
t = int(input())
a = list(map(int, input().split()))
for i in a:
    if i == 1:
        print('deficient')
        continue
    b = s(i)
    if b > i: print('abundant')
    elif b < i: print('deficient')
    else: print('perfect')